src/gc/gc_alloc.h

```diff
@@ -77,7 +77,7 @@ extern "C" inline void* gc_alloc(size_t bytes, GCKind kind_id) {
 
     // TODO This doesn't belong here (probably in PythonGCObject?)...
     if (kind_id == GCKind::PYTHON) {
-        ((Box*)r)->cls = NULL;
+        memset(r, 0, bytes);
```

Comment:
Yeah, I can get around it by explicitly null-initializing the weakref list in builtin types that opt-in.